### PR TITLE
Add embedded: prefix to --certs and use in tests

### PR DESCRIPTION
**Pull request by mberhault**
Add a small rpc.NewTestContext for testing.
Still a few tests that initialize tls configs, this will be combined later.

### CODE REVIEW

File: security/tls.go [hunk 1]
```diff
@@ -25,12 +25,18 @@ import (
...
+	"strings"
 	"sync"
 
 	"github.com/cockroachdb/cockroach/security/securitytest"
 	"github.com/cockroachdb/cockroach/util"
 )
 
+const (
+	// EmbeddedPrefix is the prefix to the --certs arg that indicates embedded certs.
+	EmbeddedPrefix = "embedded:"
```
Comment (bdarnell) [inline security/tls.go]: Gossip resolvers use "=" instead of ":"; we should probably be consistent here.

**Comment by bdarnell** [comment]
LGTM

File: security/tls.go (hunk 1 above)
Comment (MBerhault) [inline security/tls.go]: oops. fixed.

File: security/tls.go [hunk 2]
```diff
@@ -25,12 +25,18 @@ import (
...
 	"path"
+	"strings"
 	"sync"
 
 	"github.com/cockroachdb/cockroach/security/securitytest"
 	"github.com/cockroachdb/cockroach/util"
 )
 
+const (
+	// EmbeddedPrefix is the prefix to the --certs arg that indicates embedded certs.
```
Comment (spencerkimball) [inline security/tls.go]: btw, let's always refer to command line flags with a single hyphen prefix.

Comment (MBerhault) [inline security/tls.go]: Done. There and a few other places as well.

**Pull request merged**